custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -0,0 +1,256 @@
+from dateutil.relativedelta import relativedelta
+
+from corehq.apps.userreports.util import get_table_name
+from custom.icds_reports.const import AGG_SDR_TABLE
+from custom.icds_reports.utils.aggregation_helpers import  month_formatter
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import AggregationPartitionedHelper
+
+
+class AggServiceDeliveryReportHelper(AggregationPartitionedHelper):
+    helper_key = 'agg-service_delivery_report'
+    base_tablename = AGG_SDR_TABLE
+    staging_tablename = 'staging_{}'.format(AGG_SDR_TABLE)
+
+    @property
+    def monthly_tablename(self):
+        return "{}_{}".format(self.base_tablename, self.month.strftime("%Y-%m-%d"))
+
+    @property
+    def previous_agg_table_name(self):
+        return f"previous_{self.monthly_tablename}"
+
+    @property
+    def model(self):
+        from custom.icds_reports.models.aggregate import AggServiceDeliveryReport
+        return AggServiceDeliveryReport
+
+    def update_queries(self):
+        columns = (
+            ('state_id', 'awc_location_local.state_id'),
+            ('district_id', 'awc_location_local.district_id'),
+            ('block_id', 'awc_location_local.block_id'),
+            ('supervisor_id', 'awc_location_local.supervisor_id'),
+            ('awc_id', 'awc_location_local.doc_id'),
+            ('month', "'{}'".format(month_formatter(self.month))),
+            ('aggregation_level', '5'),
+            ('state_is_test','awc_location_local.state_is_test'),
+            ('district_is_test','awc_location_local.district_is_test'),
+            ('block_is_test','awc_location_local.block_is_test'),
+            ('supervisor_is_test','awc_location_local.supervisor_is_test'),
+            ('awc_is_test','awc_location_local.awc_is_test'),
+
+        )
+        yield """
+                INSERT INTO "{tmp_tablename}" (
+                    {columns}
+                )
+                (
+                SELECT
+                {calculations}
+                from awc_location_local 
+                WHERE awc_location_local.aggregation_level=5 AND awc_location_local.state_is_test<>1);
+                """.format(
+            tmp_tablename=self.staging_tablename,
+            columns=", ".join([col[0] for col in columns]),
+            calculations=", ".join([col[1] for col in columns])
+
+        ), {
+            'start_date': self.month
+        }
+
+
+        yield """
+        CREATE TABLE temp_child_table AS
```

Comment:
It would be good to drop this table before creating it, in case this step is retried part way through.